storage-operator/.gitignore

```diff
@@ -1,5 +1,3 @@
-vendor/
```

Comment:
I wouldn't remove it, TBH. *Not* having it wouldn't prevent anyone from checking things into `vendor/`. Having it requires one to add `-f` to `git add`, otherwise Git won't be willing to touch files in that folder.

Reply:
But how do they check things into `vendor`? The directory no longer exists when using 0.9.0.

Reply:
`mkdir vendor`? `go mod vendor`?...

Boils down to: having `vendor/` in here doesn't do any harm, it might even provide a benefit, so I don't see any reason to remove it.

Reply:
Fair enough, I'll add it back then.

But if we could have `vendor` here from `go mod vendor` I should keep the code to filter it then, no?

Reply:
No, because we *shouldn't* have a `vendor` directory. This is a way to ensure we don't ever get one (unless one is very stubborn).

Reply:
Ok seen with you directly: will keep the `.gitignore` entry but remove the code